core-bundle/src/Command/MigrateCommand.php

```diff
@@ -243,13 +243,37 @@ static function ($hash) use ($commandsByHash) {
 
             $count = 0;
 
-            foreach ($this->getCommandHashes($commands, 'yes, with deletes' === $answer) as $hash) {
-                $this->io->writeln(' * '.$commandsByHash[$hash]);
-                $this->installer->execCommand($hash);
-                ++$count;
-            }
+            $commandHashes = $this->getCommandHashes($commands, 'yes, with deletes' === $answer);
+
+            do {
+                $successCount = 0;
+                $exceptions = [];
+
+                foreach ($commandHashes as $key => $hash) {
+                    $this->io->write(' * '.$commandsByHash[$hash]);
+
+                    try {
+                        $this->installer->execCommand($hash);
+                        ++$count;
+                        ++$successCount;
+                        unset($commandHashes[$key]);
+                        $this->io->writeln('');
+                    } catch (\Throwable $e) {
+                        $this->io->writeln('......FAILED');
+                        $exceptions[] = $e;
+                    }
+                }
+            } while ($successCount > 0);
```

Comment:
I fail to see how this would terminate?

Reply:
It will stop, once no command could be executed successfully anymore.

Reply:
Ah my bad, I thought the variable initialization was outside the loop. :smile:  
Btw.: If the count isn't used at all I'd prefer a boolean.

Reply:
Done in aa7439ef3ef331a3b63a774aeb91804319ac81e3